src/experimental/tools/videoThumbnailLoader/thumbnail_loader.ts

```diff
@@ -0,0 +1,283 @@
+
+          const segmentLoader = createSegmentLoader(
+            loader,
+            undefined,
+            { baseDelay: 0,
+              maxDelay: 0,
+              maxRetryOffline: 0,
+              maxRetryRegular: 0, }
+          );
+
+          const segmentLoading$ = segmentLoader({
+            manifest: contentInfos.manifest,
+            period: contentInfos.manifest.periods[0],
+            adaptation: contentInfos.adaptation,
+            representation: contentInfos.representation,
+            segment,
+          });
+
+          return observableMerge(
+            bufferCleaning$.pipe(ignoreElements()),
+            segmentLoading$
+          ).pipe(
+            filter((evt): evt is { type: "data";
+                                   value: { responseData: Uint8Array }; } =>
+              evt.type === "data"),
+            mergeMap((evt) => {
+              const inventoryInfos = { manifest: contentInfos.manifest,
+                                       period: contentInfos.period,
+                                       adaptation: contentInfos.adaptation,
+                                       representation: contentInfos.representation,
+                                       segment };
+              return parser({
+                response: {
+                  data: evt.value.responseData,
+                  isChunked: false,
+                },
+                content: inventoryInfos,
+              }).pipe(
+                mergeMap((parserEvt) => {
+                  if (parserEvt.type !== "parsed-segment") {
+                    return EMPTY;
+                  }
+                  const { chunkData, appendWindow } = parserEvt.value;
+                  const segmentData = chunkData instanceof ArrayBuffer ?
+                    new Uint8Array(chunkData) : chunkData;
+                  return videoSourceBuffer
+                    .pushChunk({ data: { chunk: segmentData,
+                                         timestampOffset: 0,
+                                         appendWindow,
+                                         initSegment: null,
+                                         codec: contentInfos
+                                           .representation.getMimeTypeString() },
+                                 inventoryInfos })
+                      .pipe(map(() => {
+                        log.debug("VTL: Appended segment.", evt.value.responseData);
+                        this._videoElement.currentTime = time;
+                        return time;
+                      })
+                    );
+                })
+              );
+            })
+          );
+        }),
+        catchError((err: Error | { message?: string; toString(): string }) => {
+          const newError =
+            new VideoThumbnailLoaderError("LOADING_ERROR",
+                                          (err.message ?? err.toString()));
+          throw newError;
+        })
+      ),
+      abortError$
```

Comment:
Perhaps we could set it at first position in the `race` operator so it's more readable